### PR TITLE
Add AdvancedAction

### LINKED ISSUE #15: Make action easier usable
At the moment when giving an action to a gui you have to implement execute which isn't even run when you also implement click (which should be the default case). We somehow need to make it easy to use either one of the methods (I think the best way would be with lambdas)

### PR DESCRIPTION
**Pull request by the-codeboy**
closes #15

### CODE REVIEW

**Review by dev-null-undefined — APPROVED** [review]

File: src/main/java/ml/codeboy/bukkitbootstrap/gui/Gui.java [hunk 1]
```diff
@@ -71,6 +71,10 @@ public void onDrag(InventoryDragEvent event) {
             event.setCancelled(true);
     }
 
+    public void addItem(ItemStack item, AdvancedAction action) {
+        addItem(item, (Action) action);
+    }
+
```
Comment (dev-null-undefined) [inline src/main/java/ml/codeboy/bukkitbootstrap/gui/Gui.java:77]: I don't think that's necessary thanks to Action being super class of AdvancedAction

File: src/main/java/ml/codeboy/bukkitbootstrap/gui/MultiPageGui.java [hunk 2]
```diff
@@ -100,6 +100,10 @@ protected String getPageName(int id) {
         return "page " + id;
     }
 
+    public void addItem(ItemStack item, AdvancedAction action) {
+        addItem(item, (Action) action);
+    }
+
```
Comment (dev-null-undefined) [inline src/main/java/ml/codeboy/bukkitbootstrap/gui/MultiPageGui.java:106]: Same as with GUI I don't think we need that java class system does that automatically

**Review by dev-null-undefined — CHANGES_REQUESTED** [review]
Love the idea maybe just remove those 2 redundant functions?

**Comment by the-codeboy** [comment]
Yeah I didn't add those at first however if those don't exist you can not use lambdas since you can not use lambdas for sub classes

**Comment by dev-null-undefined** [comment]
> Yeah I didn't add those at first however if those don't exist you can not use lambdas since you can not use lambdas for sub classes

ok well then yep this looks ready to be merged

**Review by dev-null-undefined — COMMENTED** [review]
Looks good well done

**Review by dev-null-undefined — APPROVED** [review]

**Pull request merged**